src/pkg/backup/backup.go

```diff
@@ -111,12 +114,31 @@ func New(
 		}
 	}
 
+	ctx = clues.Add(ctx, "is_assist_backup", isAssistBackup)
+	ctx = clues.Add(ctx, "error_count", errCount)
+	ctx = clues.Add(ctx, "skip_count", skipCount)
+
+	tags := map[string]string{
+		model.ServiceTag: selector.PathService().String(),
+	}
+
+	// Add tags to mark this backup as either assist or merge. This is used to:
+	// 1. Filter assist backups by tag during base selection process
+	// 2. Differentiate assist backups from merge backups
+	if isAssistBackup {
+		tags[model.AssistBackupTag] = ""
+	} else if errCount == 0 {
```

Comment:
The mix of ownership here worries me a bit.  This is both externally asking the caller to evaluate the state of the backup, and internally asserting conditions that may or may not conflict with the caller's decision.  I feel like one or the other is best for long term maintenance: let the caller tell you whether its an assist or a merge, or make that decision on their behalf.